lib/dslash_pack2.cu

```diff
@@ -44,17 +45,22 @@ namespace quda {
 
     const DslashConstant dc; // pre-computed dslash constants for optimized indexing
 
+    real a; // preconditioned twisted-mass scaling parameter
+    real b; // preconditioned twisted-mass twist factor
+    bool twist; // whether we are doing preconditioned twisted-mass or not
+
     int_fastdiv threads;
     int threadDimMapLower[4];
     int threadDimMapUpper[4];
 
     int_fastdiv swizzle;
     int sites_per_block;
 
-    PackArg(void **ghost, const ColorSpinorField &field, int nFace, bool dagger, int parity)
+    PackArg(void **ghost, const ColorSpinorField &field, int nFace,
+            bool dagger, int parity, double a, double b)
       : field(field, 1, nullptr, nullptr, reinterpret_cast<Float**>(ghost)),
         nFace(nFace), dagger(dagger), parity(parity), nParity(field.SiteSubset()),
-        dc(field.getDslashConstant())
+        dc(field.getDslashConstant()), a(a), b(b), twist(a != 0.0 && b != 0.0)
```

Comment:
Should this be `a != 0.0 || b != 0.0`? I imagine it doesn't make sense to call this op without having both non-zero, but still, it could happen, and it at least should throw a warning/error (assuming it doesn't already somewhere...)

Reply:
Fixed 78a9149